AlphaWallet/Transfer/Coordinators/SignMessageCoordinator.swift

```diff
@@ -5,35 +5,80 @@ import UIKit
 import AlphaWalletFoundation
 
 protocol SignMessageCoordinatorDelegate: AnyObject {
-    func coordinator(_ coordinator: SignMessageCoordinator, didSign result: Swift.Result<Data, KeystoreError>)
+    func coordinator(_ coordinator: SignMessageCoordinator, didSign result: Data)
     func didCancel(in coordinator: SignMessageCoordinator)
 }
 
+struct SignMessageValidator {
+
+    enum ValidatorError: Error {
+        case emptyMessage
+        case chainIdNotMatches(activeServer: RPCServer, server: RPCServer, source: Analytics.SignMessageRequestSource)
+    }
+
+    let server: RPCServer
+    let source: Analytics.SignMessageRequestSource
+
+    func validate(mesage: SignMessageType) throws {
```

Comment:
Let's keep this for now. But I'm on the fence on this: should we return bools for such validations or should we throw?

pros for throw
---
* more than 2 values (but we can return enums or optional enums but by then, throwing seems obviously better)
* keep the positive path local. More consistent with railway/reactive

pros for bool
---
* clearer and local
* easier to be sure we handle it

@oa-s any thoughts and more pros and cons?